### PR TITLE
Fix #1984 CLuaFunctionRef not working in new parser

### LINKED ISSUE #1984: New parser - CLuaFunctionRef not working
**Describe the bug**
You can't use CLuaFunctionRef in new parser

**To reproduce**
Make one of argument of type CLuaFunctionRef 

**Expected behaviour**
Don't crash

**Screenshots**
https://cdn.discordapp.com/attachments/366384007535001612/793898701638598706/unknown.png

**Version**
newstest

**Additional context**
/

### PR DESCRIPTION
**Pull request by sbx320**
Turns out `luaM_toref` adjusts the stack top, breaking everything.

### CODE REVIEW

File: Shared/mods/deathmatch/logic/lua/CLuaFunctionParser.h [hunk 1]
```diff
@@ -444,7 +444,33 @@ struct CLuaFunctionParserBase
...
+            CLuaMain* pLuaMain = g_pClientGame->GetLuaManager()->GetVirtualMachine(L);
+        #else
+            CLuaMain* pLuaMain = g_pGame->GetLuaManager()->GetVirtualMachine(L);
+        #endif
+            const void* pFuncPtr = lua_topointer(L, index);
+
+            if (CRefInfo* pInfo = MapFind(pLuaMain->m_CallbackTable, pFuncPtr))
+            {
+                // Re-use the lua ref we already have to this function
+                pInfo->ulUseCount++;
```
Comment (Pirulax) [inline Shared/mods/deathmatch/logic/lua/CLuaFunctionParser.h:457]: Is it decreased?

File: Shared/mods/deathmatch/logic/lua/CLuaFunctionParser.h [hunk 2]
```diff
@@ -444,7 +444,33 @@ struct CLuaFunctionParserBase
...
+                ++index;
```
Comment (Pirulax) [inline Shared/mods/deathmatch/logic/lua/CLuaFunctionParser.h:458]: This is repeated in the `else` branch below. It could be moved to before the `if`.

Comment (sbx320) [reply Shared/mods/deathmatch/logic/lua/CLuaFunctionParser.h:458]: The else branch uses index pre-increment, so we can't move that up (at least without subtracting 1)

File: Shared/mods/deathmatch/logic/lua/CLuaFunctionParser.h (hunk 1 above)
Comment (sbx320) [reply Shared/mods/deathmatch/logic/lua/CLuaFunctionParser.h:457]: Yeah, in the destructor of CLuaFunctionRef

See https://github.com/multitheftauto/mtasa-blue/blob/a0410c68945b6b3d9225c8c90af2bb6b1cda5eaa/Client/mods/deathmatch/logic/lua/CLuaFunctionRef.cpp#L115

**Review by Pirulax — APPROVED** [review]
lgtm

**Comment by Pirulax** [comment]
I presume you have tested it 😆
(Have you?)

**Pull request merged**